pandas/core/internals/blocks.py

```diff
@@ -244,6 +244,8 @@ def make_block(self, values, placement=None) -> "Block":
         """
         if placement is None:
             placement = self.mgr_locs
+        if self.is_extension:
```

Comment:
my point is that is this necessary this (if), or are you saying that this is a perf issue? (e.g. we don't want to *always* call _block_shape

Reply:
ATM we are calling _block_shape in a bunch of places before calling `ExtensionBlock.make_block`.  This is just moving all of those calls to one place

Reply:
> we don't want to _always_ call `_block_shape`

With some corner-case-caveats, we only want to call it if self.is_extension, and we pretty much _always_ have to call it if self.is_extension (otherwise we could use make_block_same_class